packages/plugins/transaction-logger/src/index.js

```diff
@@ -247,27 +259,42 @@ export default class TransactionLogger {
     this.embark.registerAPICall(
       'get',
       apiRoute,
-      (req, res) => {
-        res.send(JSON.stringify(this._getLogs()));
+      async (req, res) => {
+        res.send(JSON.stringify(await this._getLogs()));
       }
     );
   }
 
-  _getLogs() {
-    const data = this._readLogs();
+  async _getLogs() {
+    const data = await this._readLogs();
     return Object.values(data).reverse();
   }
 
   _saveLog(log) {
     this.writeLogFile.push(log);
   }
 
-  _readLogs() {
-    this.fs.ensureFileSync(this.logFile);
-    try {
-      return JSON.parse(this.fs.readFileSync(this.logFile));
-    } catch (_error) {
-      return {};
-    }
+  async _readLogs() {
+    return new Promise(async resolve => {
```

Comment:
I think the promise inside the async function isn't needed since it always returns a promise.

```js
async _readLogs() {
  try {
    await this.fs.ensureFile(this.logFile);
    let data = await this.fs.readFile(this.logFile);

    data = data.toString();

    if (!data) {
      return {};
    }

    // remove last comma and add braces around
    data = `{${data.substring(0, data.length - 2)}}`;

    return JSON.parse(data);
  } catch (error) {
    this.logger.error('Error reading contract log file', error.message);
    this.logger.trace(error.trace);
    return {};
  }
}
```

Reply:
Good point, I forgot to remove it